### ISSUE TITLE
 Login, logout ,signup backend 

### ISSUE THREAD

**Issue by ChandraST12** [SwanandD121/FeatherPerfect_fe#43]
I would like to contribute by implementing the **login, logout, and signup functionality with authorization using Node.js** for your project.

 It would be great if you could assign me this issue under the **GSSoC-ext** and **Hacktoberfest** labels. I am excited about the opportunity to work on this and look forward to contributing.

**Comment by SwanandD121** [comment]
@ChandraST12 go ahead.
First explain the approach on the project post on discord.

**Comment by ChandraST12** [comment]
@SwanandD121 Could you please share your Discord so I can explain my approach?

**Comment by ChandraST12** [comment]
> **Project Structure:**
> Create a server folder: This folder will contain all the backend-related files for your application, including models, controllers, routes, middleware, and utility files.
> 
> **Models:**
> 
> > **User Model**: This will handle user-related information such as firstName, lastName, email, password, accountType,  etc.
> > **OTP Model:** This will store OTPs generated for user registration or password reset purposes, with fields for the email and otp.

> **Middleware**:
> 
> **Authorization Middleware:** To protect certain routes and ensure that users are authenticated, implement JWT (JSON Web Token) authentication. This middleware will check the validity of the JWT token for protected routes like changePassword.
> Controllers: The business logic of the application will be placed in controllers. These functions will be responsible for:
> 
> **Sign Up:**
> Validate input fields.
> Check if the user already exists.
> Verify the OTP sent to the user.
> Hash the password using bcrypt and create a new user.

> **Login:**
> Authenticate the user by verifying the email and password.
> Generate and return a JWT token for future authentication.

> **Change Password:**
> Validate the old password and ensure the new password matches the confirmation password.
> Hash and update the new password in the database.

> **Send OTP:**
> Generate a unique OTP using the otp-generator library.
> Ensure the OTP is unique and store it in the OTP model.
> Respond with the generated OTP or send it via email.
> Routes: Routes will define the endpoints for interacting with the backend and map to the appropriate controller functions:
> 
> POST /auth/signup: For user registration.
> POST /auth/login: For logging in users.
> POST /auth/send-otp: To generate and send OTP for registration or password recovery.
> PUT /auth/change-password: For users to change their password (JWT protected).
> 
> 



Folder Structure 

/server
  ├── controllers
  │   ├── authController.js     # Handles login, signup, change password
  │   ├── otpController.js      # Handles OTP generation and validation
  ├── models
  │   ├── User.js               # User model schema
  │   ├── OTP.js                # OTP model schema
  │   
  ├── middleware
  │   └── auth.js               # JWT authorization middleware
  ├── routes
  │   ├── authRoutes.js         # Authentication-related routes
  ├── utils
  │   └── mailSender.js         # Utility for sending emails
  ├── index.js                  # Main server file
  └── .env                      # Environment variables (JWT secret, etc.)

**Comment by SwanandD121** [comment]
> @SwanandD121 Could you please share your Discord so I can explain my approach?

@swanand_in

**Comment by ChandraST12** [comment]
@chandra21
Yes, I would appreciate it if we could discuss in more depth about how to start working on backend

**Comment by ParasY1724** [comment]
@ChandraST12 can i take this task ?

**Comment by tarunkumar2005** [comment]
@SwanandD121 I see the issue is not solved so you can assign this to me i am more experienced and professional you can assign this to me i will quickly resolve your issue.

**Comment by SaanyaGarg5** [comment]
you can assign this task to me..
